modules/internal/Bytes.chpl

```diff
@@ -416,6 +428,14 @@ module Bytes {
       return getCStr(this);
     }
 
+    pragma "no doc"
+    inline proc param c_str() param : c_string {
+      inline proc _cast(type t:c_string, x:string) {
+        return __primitive("cast", t, x);
+      }
```

Comment:
I'm not seeing how/why the above inner function could apply?

Reply:
huh.. good catch. In the test, I am assigning the return value of `c_str` on a param `bytes` to a `var`. So, maybe the non-param `c_str` was being called instead of this. I'll fix it and make sure that it is being called.

Reply:
My assumption was wrong. Apparently that inner function is not even necessary any more for the cast to resolve. I did some basic tests on the string's corresponding method and that also can be removed. I'll do that later on.